templates/tfengine/components/cicd/main.tf

```diff
@@ -250,14 +249,51 @@ resource "google_project_iam_member" "cloudbuild_scheduler_sa_project_iam" {
 }
 {{- end}}
 
+{{- if not (has . "service_account")}}
+resource "google_service_account" "cloudbuild_sa" {
+  project      = var.project_id
+  account_id   = "cloudbuild-sa"
```

Comment:
Instead of hardcoding the ID here, we can probably just ask users to provide the ID and we always create it in the `devops` project in the cicd recipe. Or another option is instead of asking for a service_account_id, we ask for a boolean value `use_custom_service_account` or something like that. And based on that value, we create or do not create a new SA to be used for cloud build jobs. In that case, we hardcode a value such as `cloudbuild-sa`. The first option is more flexible but second option seems fine to me as well.

Also if users do not provide service_account, then this block of config will be executed, however, in the other places, it's still using `var.service_account_email`. I guess you are relying on the default value of it declared in the `variables.tf`. That's a bit hacky. Also the other resources that use this SA depends on the SA to be created first so reference should be used.